### PR TITLE
Update XDR for stellar-core v10.0.0 (CAP-0003)

### PR DESCRIPTION
**Pull request by nebolsin**
- [x] Update XDR definitions for stellar-core v10.0.0 (introduces Liabilities and other changes to support asset-backed offers as per [CAP-0003 Specification](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0003.md#specification))
- [x] Backfill `CHANGELOG.md` with `0.16.0` and `0.17.0` changes

/cc @nullstyle

### CODE REVIEW

**Comment by nebolsin** [comment]
@bartekn, @nullstyle could you please review this PR? It would be nice to have full support for protocol 10 in ruby-stellar-base when public network upgrades.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,21 @@ file.  This project adheres to [Semantic Versioning](http://semver.org/).
 As this project is pre 1.0, breaking changes may happen for minor version
 bumps.  A breaking change will get clearly notified in this log.
 
-## [unreleased](https://github.com/stellar/ruby-stellar-base/compare/v0.15.0...master)
+## [unreleased](https://github.com/stellar/ruby-stellar-base/compare/v0.17.0...master)
+### Added
+- Update XDR definitions for stellar-core v10.0.0rc2 (introduces Liabilities and other changes to support asset-backed offers as per [CAP-0003 Specification](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0003.md#specification))
```
Comment (bartekn) [inline CHANGELOG.md]: Please change `v10.0.0rc2` to `v10.0.0`.

**Review by bartekn — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (nebolsin) [reply CHANGELOG.md]: Fixed, thank you

**Comment by nebolsin** [comment]
@bartekn Thank you for review, the issue you mentioned is fixed.

**Review by bartekn — APPROVED** [review]

**Pull request merged**